posthog/hogql_queries/insights/trends/trends_query_runner.py

```diff
@@ -129,23 +132,68 @@ def to_actors_query(self) -> ast.SelectQuery | ast.SelectUnionQuery:
     def calculate(self):
         queries = self.to_query()
 
-        res = []
-        timings = []
+        res_matrix: List[List[Any] | Any | None] = [None] * len(queries)
+        timings_matrix: List[List[QueryTiming] | None] = [None] * len(queries)
+        errors: List[Exception] = []
 
-        for index, query in enumerate(queries):
-            series_with_extra = self.series[index]
+        def run(index: int, query: ast.SelectQuery | ast.SelectUnionQuery, is_parallel: bool):
+            try:
+                series_with_extra = self.series[index]
 
-            response = execute_hogql_query(
-                query_type="TrendsQuery",
-                query=query,
-                team=self.team,
-                timings=self.timings,
-                modifiers=self.modifiers,
-            )
+                response = execute_hogql_query(
+                    query_type="TrendsQuery",
+                    query=query,
+                    team=self.team,
+                    timings=self.timings,
+                    modifiers=self.modifiers,
+                )
 
-            timings.extend(response.timings)
+                timings_matrix[index] = response.timings
+                res_matrix[index] = self.build_series_response(response, series_with_extra, len(queries))
+            except Exception as e:
+                errors.append(e)
+            finally:
+                if is_parallel:
+                    from django.db import connection
+
+                    connection.close()
```

Comment:
I think this could get an extra comment explaining that this won't kill the app

Reply:
Yep, fair, will add